### PR TITLE
Backport of MAGETWO-59685 for Magento 2.1 - Checkout pages very slow …

### PR DESCRIPTION
**Pull request by hostep**
…due to pulling in JSON config, remove unneeded region definitions from the checkout page. #6997 #4868

### Description
This is a backport of issue MAGETWO-59685 for Magento 2.1

### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/6997: Remove unneeded region definitions from the /checkout page.
2. Partly https://github.com/magento/magento2/issues/4868: Checkout page very large and quite slow.

Reference to commit on the develop branch: https://github.com/magento/magento2/commit/6a5ece2f27934b2ddf7a939617c12507d264ee90

### Discussion
Together with the [backport of MAGETWO-60351](https://github.com/magento/magento2/pull/9365), this significantly speeds up the checkout process, even up to multiple seconds. This is due to the json data structure being outputted on the checkout containing way too many elements then necessary, which makes the client side processing of this json take a very long time (sometimes up to 10 seconds). In a concrete example of one of our webshops, this reduced the json structure from 175867 lines to 4003 lines.

MAGETWO-59685 in particular fixes not outputting country and region data for countries to which the webshop isn't selling.

### CODE REVIEW

File: app/code/Magento/Checkout/Block/Checkout/DirectoryDataProcessor.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Copyright © 2016 Magento. All rights reserved.
```
Comment (omiroshnichenko) [inline app/code/Magento/Checkout/Block/Checkout/DirectoryDataProcessor.php]: Update copyright like in current 2.1-develop.

File: app/code/Magento/Checkout/Test/Unit/Block/Checkout/DirectoryDataProcessorTest.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Copyright © 2016 Magento. All rights reserved.
```
Comment (omiroshnichenko) [inline app/code/Magento/Checkout/Test/Unit/Block/Checkout/DirectoryDataProcessorTest.php]: Same problem with copyright.

**Comment by hostep** [comment]
@omiroshnichenko: I just amended the commit with the changed copyright header, and force pushed it.

**Comment by magento-team** [comment]
@hostep thank you for your contribution. Your Pull Request has been successfully merged

**Pull request merged**